python/src/ngraph/utils/input_validation.py

```diff
@@ -33,3 +34,104 @@ def assert_list_of_ints(value_list, message):  # type: (Iterable[int], str) -> N
     except TypeError:
         log.warning(message)
         raise UserInputError(message, value_list)
+
+
+def _check_value(op_name, attr_key, value, val_type, cond=None):
+    # type: (str, str, Any, Type, Optional[Callable[[Any], bool]]) -> bool
+    """Check whether provided value satisfies specified criteria.
+
+    :param      op_name:        The operator name which attributes are checked.
+    :param      attr_key:       The attribute name.
+    :param      value:          The value to check.
+    :param      val_type:       Required value type.
+    :param      cond:           The optional function running additional checks.
+
+    :raises     UserInputError:
+    :return:    True if attribute satisfies all criterias. Otherwise False.
+    """
+    if not np.issubdtype(type(value), val_type):
```

Comment:
I'm not sure the type checking here will work as you expect.
```
>>> a = np.array([123], dtype=np.int32)
>>> np.issubdtype(type(a), np.int32)
False
```

Reply:
The `_check_value` function is intended to be used only with scalar values and _internally_ it is so. Moreover this function is _hidden_ since it's name starts with an underscore. Do you want to enforce scalar input in this function?